types/src/database_sql.rs

```diff
@@ -305,10 +305,26 @@ impl Database {
         todo!("Implement image annotations")
     }
 
-    pub async fn update_word(&self, word: &AnnotatedFormUpdate) -> Result<Uuid> {
-        query_file!("queries/update_word.sql", word.id, word.source.value())
-            .execute(&self.client)
-            .await?;
+    pub async fn update_word(&self, word: AnnotatedFormUpdate) -> Result<Uuid> {
+        let mut source: Vec<Option<String>> = Vec::new();
+        let mut commentary: Vec<Option<String>> = Vec::new();
+
+        if !word.source.is_undefined() {
+            source.push(word.source.take());
+        }
+
+        if !word.commentary.is_undefined() {
+            commentary.push(word.commentary.take());
+        }
```

Comment:
We could remove mutability here by doing it this way instead:

```rs
let source = if word.source.is_undefined() {
  Vec::new()
} else {
  vec![word.source.take()]
};
```

Reply:
Since we're gonna use this pattern a bunch, you could also abstract this into a function that goes from `MaybeUndefined<T>` -> `Vec<Option<T>>`